### PR TITLE
spice-gtk 0.36 (new formula)

### PR DESCRIPTION
**Pull request by jeffreywildman**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/spice-gtk.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  depends_on "opus"
+  depends_on "pango"
+  depends_on "pixman"
+  depends_on "spice-protocol"
+  depends_on "usbredir"
+
+  def install
+    # Some files (vncdisplaykeymap.c) require building as Objective-C
+    # https://www.mail-archive.com/spice-devel@lists.freedesktop.org/msg40085.html
+    ENV["CFLAGS"] = "-ObjC -g -O2"
```
Comment (fxcoudert) [inline Formula/spice-gtk.rb]: - `-g -O2` are handled by Homebrew, do not add those.
- `-ObjC` should be given by the configure/Makefile process, really. Has this bug been reported?

File: Formula/spice-gtk.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                   "-I#{Formula["gtk+3"].include}/gtk-3.0",
+                   "-I#{Formula["pango"].include}/pango-1.0",
+                   "-I#{Formula["spice-protocol"].include}/spice-1",
+                   "-I#{include}/spice-client-glib-2.0",
+                   "-I#{include}/spice-client-gtk-3.0",
+                   "-L#{lib}",
+                   "-lspice-client-glib-2.0",
+                   "-lspice-client-gtk-3.0",
+                   "-o", "test"
+    system "./test"
```
Comment (fxcoudert) [inline Formula/spice-gtk.rb]: Great test!

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/spice-gtk.rb (hunk 1 above)
Comment (jeffreywildman) [reply Formula/spice-gtk.rb]: Reported [upstream](https://gitlab.freedesktop.org/spice/spice-gtk/issues/88) with proposed patch. Does patch need to be accepted upstream before applying to formula?

File: Formula/spice-gtk.rb (hunk 2 above)
Comment (jeffreywildman) [reply Formula/spice-gtk.rb]: Thanks!

File: Formula/spice-gtk.rb (hunk 1 above)
Comment (jeffreywildman) [reply Formula/spice-gtk.rb]: Upstream reported/pushed issue to a dependency, gtk. Issue [here](https://gitlab.gnome.org/GNOME/gtk/issues/1737). Patch and merge request [here](https://gitlab.gnome.org/GNOME/gtk/merge_requests/641).

Once gtk+3 is updated, then this formula can be updated.

**Pull request closed without merging**